homeassistant/components/homewizard_energy/__init__.py

```diff
@@ -0,0 +1,98 @@
+"""The Homewizard Energy integration."""
+import asyncio
+import logging
+
+import aiohwenergy
+import async_timeout
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_API, COORDINATOR, DOMAIN, PLATFORMS
+from .coordinator import HWEnergyDeviceUpdateCoordinator as Coordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Homewizard Energy from a config entry."""
+
+    _LOGGER.debug("__init__ async_setup_entry")
+
+    # Get api and do a initialization
+    energy_api = aiohwenergy.HomeWizardEnergy(entry.data.get(CONF_IP_ADDRESS))
+
+    # Validate connection
+    initialized = False
+    try:
+        with async_timeout.timeout(10):
+            await energy_api.initialize()
+            initialized = True
+
+    except (asyncio.TimeoutError, aiohwenergy.RequestError) as ex:
+        _LOGGER.error(
+            "Error connecting to the Energy device at %s",
+            energy_api.host,
+        )
+        raise ConfigEntryNotReady from ex
+
+    except aiohwenergy.DisabledError as ex:
+        _LOGGER.error("API disabled, API must be enabled in the app")
+        raise ConfigEntryNotReady from ex
```

Comment:
Return `False` because the config entry will not be able to set up and we should stop retrying. Users can reload the entry if they have re-enabled the API.
```suggestion
        return False
```

Reply:
I try to implement this in while moving the setup to coordinator (requested in https://github.com/home-assistant/core/pull/55812#discussion_r763643640). How can I tell coordinator/HA to stop retrying when it detected a disabled API?

Reply:
raise a specific exception that you catch when you call first_refresh

Reply:
The coordinator raises an `ConfigEntryNotReady` when something when wrong https://github.com/home-assistant/core/blob/af603d04272ec489d03b26cb6384d062611d0a75/homeassistant/helpers/update_coordinator.py#L159-L161
I couldn't figure out how to catch the original exception and can't find a component that tries to. To get the desired behaviour I tried it like this; https://github.com/home-assistant/core/pull/55812/commits/e06bb269948a1073d6bea8f3b240e6c8587aab29. What do you think?